src/isar/storage/slimm_storage.py

```diff
@@ -0,0 +1,103 @@
+import json
+import logging
+
+from azure.identity import DefaultAzureCredential
+from injector import inject
+from requests import RequestException
+from requests_toolbelt import MultipartEncoder
+
+from isar.config import config
+from isar.models.mission_metadata.mission_metadata import MissionMetadata
+from isar.services.auth.azure_credentials import AzureCredentials
+from isar.services.service_connections.request_handler import RequestHandler
+from isar.storage.storage_interface import StorageException, StorageInterface
+from isar.storage.utilities import get_filename, get_inspection_type
+from robot_interface.models.inspection.inspection import Inspection
+
+
+class SlimmStorage(StorageInterface):
+    @inject
+    def __init__(self, request_handler: RequestHandler) -> None:
+        self.request_handler: RequestHandler = request_handler
+        self.logger = logging.getLogger("uploader")
+
+        self.credentials: DefaultAzureCredential = (
+            AzureCredentials.get_azure_credentials()
+        )
+
+        self.client_id: str = config.get("service_connections", "slimm_client_id")
+        self.scope: str = config.get("service_connections", "slimm_app_scope")
+        self.request_scope: str = f"{self.client_id}/{self.scope}"
+
+        self.url: str = config.get("service_connections", "slimm_api_url")
+
+    def store(self, inspection: Inspection, metadata: MissionMetadata):
```

Comment:
The store function currently does a lot. Would it make sense to split it out into supporting functions? Something like: _convert_to_slimm_format and _upload_to_slim?

Reply:
Done